keras_cv/layers/preprocessing/random_shear.py

```diff
@@ -151,3 +169,168 @@ def get_config(self):
             }
         )
         return config
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation, **kwargs):
+        image = None
+        image = kwargs.get("image")
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomShear()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomShear(bounding_box_format='xyxy')`"
+            )
+        height, width, _ = image.shape
+
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+            dtype=self.compute_dtype,
+        )
+        x, y = transformation
+        # apply horizontal shear
+        if x is not None:
+            bounding_boxes = self._apply_horizontal_transformation_to_bounding_box(
+                bounding_boxes, x
+            )
+        # apply vertical shear
+        if y is not None:
+            bounding_boxes = self._apply_vertical_transformation_to_bounding_box(
+                bounding_boxes, y
+            )
+        # clip bounding boxes value to 0-image height and 0-image width
+        bounding_boxes = self._clip_bounding_box(height, width, bounding_boxes)
+        bounding_boxes = keras_cv.bounding_box.convert_format(
+            bounding_boxes,
+            source="xyxy",
+            target=self.bounding_box_format,
+            images=image,
+            dtype=self.compute_dtype,
+        )
+        return bounding_boxes
+
+    def _clip_bounding_box(self, height, width, bounding_boxes):
+        """clips bounding boxes b/w 0 - image width and 0 - image height"""
+        x1, y1, x2, y2 = tf.split(bounding_boxes, 4, axis=1)
```

Comment:
you're dropping the rest of the axes here.  Check out how we do split in the converter functions and copy this pattern.

Reply:
Understood.